newIDE/app/src/UI/EditorMosaic/CloseButton.js

```diff
@@ -24,6 +28,7 @@ export default class CloseButton extends Component {
             {({ mosaicWindowActions }) => (
               <IconButton
                 onClick={() => {
+                  if (this.props.closeActions) this.props.closeActions();
                   mosaicActions.remove(mosaicWindowActions.getPath());
```

Comment:
See in `newIDE\app\src\UI\EditorMosaic\index.js` you have the current stuff for add a node in mosaic with `addNode`.
It would be better to add a similar for removeNode. not sure of the best way to do it thought.

Reply:
React recommends toggling of components through react state. As you can see , Material-UI also follows this convention. This way we can use this state anywhere in component and other components.  
`mosiacAction.remove(path)` is already doing the job of removal of node.
The code in `componentWillMount` I wrote only for the first render. That is when project loads for the first time.
By doing this we will set the sceneEditor state of the panels as they were before closing the project.

Reply:
React recommends toggling of components through react state. As you can see , Material-UI also follows this convention. This way we can use this state anywhere in component and other components.  
`mosiacAction.remove(path)` is already doing the job of removal of node.
The code in `componentWillMount` I wrote only for the first render. That is when project loads for the first time.
By doing this we will set the sceneEditor state of the panels as they were before closing the project.

Another example of state use is as you can see here , we have used state to display message : 
https://github.com/4ian/GDevelop/blob/98a1d69489c5410d28617f11686351e093157c5d/newIDE/app/src/SceneEditor/index.js#L1208-L1216

So I don't think removing state is a good idea in case of toggle components. But You gave a solid idea of creating this `removeNode` function. Now speed of toggling of panels can be reduced by half. 
Thank you for your hint. In the next commit I will apply these changes.

Reply:
> React recommends toggling of components through react state.
So I didn't removed the code in the `componentWillMount` method. Although app works fine without including it but the state of `SceneEditor` should be according to the state of panels, so if we need the state of the panels somewhere else it will not create any conflicts.
The code now works with shortcuts also.
@Bouh Please guide me if some refactors or updates are needed ? ;)